File: testdata/jennies/builders/dataquery_variant_builder/PythonBuilder/builders/dataquery_variant_builder.py
import typing
from ..cog import builder as cogbuilder
from ..models import dataquery_variant_builder


class LokiBuilder(cogbuilder.Builder[dataquery_variant_builder.Loki]):
    _internal: dataquery_variant_builder.Loki

    def __init__(self):
        self._internal = dataquery_variant_builder.Loki()

    def build(self) -> dataquery_variant_builder.Loki:
        """
        Builds the object.
        """
        return self._internal    
    
    def expr(self, expr: str) -> typing.Self:    
        self._internal.expr = expr
    
        return self
    

File: testdata/jennies/builders/dataquery_variant_builder/RustBuilder/src/builders/dataquery_variant_builder.rs
use crate::cog;
use crate::cog::variants;
use crate::types::dataquery_variant_builder;

#[derive(Debug, Clone)]
pub struct LokiBuilderBuilder {
    internal: dataquery_variant_builder::Loki,
    pub(crate) errors: Vec<cog::BuildError>,
}

impl LokiBuilderBuilder {
    pub fn new() -> Self {
        Self {
            internal: dataquery_variant_builder::Loki::default(),
            errors: Vec::new(),
        }
    }
}

impl Default for LokiBuilderBuilder {
    fn default() -> Self {
        Self::new()
    }
}

impl LokiBuilderBuilder {
    pub fn expr(mut self, expr: String) -> Self {
        self.internal.expr = expr;

        self
    }
}

impl cog::Builder<Box<dyn variants::Dataquery>> for LokiBuilderBuilder {
    fn build(&self) -> Result<Box<dyn variants::Dataquery>, Vec<cog::BuildError>> {
        if !self.errors.is_empty() {
            return Err(self.errors.clone());
        }

        Ok(Box::new(self.internal.clone()))
    }
}
